### ISSUE TITLE
 New "app.kubernetes.io/version" label causes "must be no more than 63 characters" error on reconcile

### ISSUE THREAD

**Issue by ringerc** [open-telemetry/opentelemetry-operator#879]
PR https://github.com/open-telemetry/opentelemetry-operator/pull/797 added `app.kubernetes.io/version`  to `metadata.labels` . The label value is computed from the image tag.

If the opentelemetry operator is deployed using an image specified by a sha256 hash, this breaks reconciliation with errors like

> failed to reconcile config maps","error":"failed to reconcile the expected configmaps: failed to apply changes: ConfigMap \"otel-collector\" is invalid: metadata.labels: Invalid value: \"8f65b4d94bb5290c8fc1540703c06f7a7a12cfd917d2f141bdc8a18803828615\": must be no more than 63 characters"

It is incorrect to parse the image tag to get the version of the component. Instead, the version should be injected as a `--build-arg` / `ARG` into the container, then:

* propagated to an `ENV` var, and the code should use the version from the environment; or
* passed to `go build` to set constants in the build per https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications

https://github.com/open-telemetry/opentelemetry-operator/pull/797 introduced a breaking regression.

**Comment by ringerc** [comment]
Workaround:

Downgrade your `opentelemetry-operator` Kustomizations and (if overridden from upstream kustomizations) your container image to v0.48.0.

Find and delete all Deployments created by the v0.50.0 operator, which incorrectly add a label to selector, resulting in reconciliation errors on downgrade per #840:

```
kubectl delete --all-namespaces deployment -l app.kubernetes.io/managed-by=opentelemetry-operator --dry-run=client
```

Omit the `--dry-run=client` to really execute the delete.

**Comment by pavolloffay** [comment]
This was resolves in #849 

please re-open if the issue is still lthere.

**Comment by ringerc** [comment]
Thanks. I didn't land up working out how to write a kuttl test for it, sorry.
